diesel/src/query_source/mod.rs

```diff
@@ -48,3 +48,13 @@ pub trait Table: QuerySource + AsQuery + Sized {
     fn primary_key(&self) -> Self::PrimaryKey;
     fn all_columns() -> Self::AllColumns;
 }
+
+pub trait AppearsInFromClause<QS> {
```

Comment:
ℹ️ Tracks how many times a table occurs in QuerySource (from/join clause)

Maybe add ↑ as doc comment?